File: python/leetcode/maximum-product-subarray.py
class Solution:
    def maxProduct(self, nums):
        start_idx = 0
        max_p = nums[start_idx]
        while (start_idx < len(nums)):
            this_product = nums[start_idx]
            if this_product > max_p:
                max_p = this_product
            for i in range(start_idx + 1, len(nums)):
                this_product *= nums[i]
                if this_product > max_p:
                    max_p = this_product
            start_idx += 1
        return max_p

File: java/leetcode/maximum-product-subarray.java
class Solution {
    static int maxProduct(int[] nums) {
    	int startIdx = 0;
    	int maxP = nums[startIdx];
    	int thisProduct = maxP;
    	int i = 0;
    	while (startIdx < nums.length) {
    		thisProduct = nums[startIdx];
    		if (thisProduct > maxP) {
    			maxP = thisProduct;
    		}
    		for (i = startIdx + 1 ; i < nums.length; i++) {
    			thisProduct *= nums[i];
    			if (thisProduct > maxP) {
    				maxP = thisProduct;
    			}
    		}
    		startIdx++;
    	}
    	return maxP;
    }
}
